doc/unison-manual.tex

```diff
@@ -1998,16 +1998,18 @@
 synchronization complete normally, press ``Ctrl-D'' or send signal
 \verb|SIGUSR2| instead.
 Closing the input, receiving an EOF, or receiving a \verb|^D| (0x04) from the
-input terminal or a redirected standard input all have the same effect. If the
-standard input is redirected from a regular file, is not open or is not open
-for reading already at Unison startup then it is ignored and only signals can
-be used to send the stop request.
+input terminal or a redirected standard input all have the same effect. EOF is
+not interpreted as a stop request when standard input is redirected from a
+regular file. If the standard input is not open or is not open for reading
+already at Unison startup then it is ignored and only signals can be used to
+send the stop request.
```

Comment:
This seems to have some idea that the regular file will be written, and that unison will be doing rewind and seeing the content again.

And, if there is an empty/32byte rule, it should be in the manual.   Perhaps the manual should say that it must  be a 0-byte file and stopping must write a single 0x4, if that's how it is, or maybe 0x4\n is ok too.  The specification of what to do should be narrow.    We also need to avoid UTF-8/encoding issues.

Reply:
> This seems to have some idea that the regular file will be written, and that unison will be doing rewind and seeing the content again.

Need to improve the language. That's actually not what will happen; there will be no rewind, only appended content is read. (In fact, I thought about rewinding but it seems too much...)

> And, if there is an empty/32byte rule, it should be in the manual. Perhaps the manual should say that it must be a 0-byte file and stopping must write a single 0x4, if that's how it is, or maybe 0x4\n is ok too. The specification of what to do should be narrow. We also need to avoid UTF-8/encoding issues.

Can add the 0-byte rule. Currently (can be changed, of course), only 0x4 is checked but any other input before and after that byte is ignored. So, 0x4\n will work but so will \n0x4. Encodings will cause any issues here because it's just the raw byte 0x4 that is checked.

Reply:
So read(2) will return EOF, and then when a byte is written a further read(2) call with return that byte?  Is that guaranteed?